compiler/rustc_middle/src/query/mod.rs

```diff
@@ -44,8 +44,8 @@ rustc_queries! {
     ///
     /// This can be conveniently accessed by `tcx.hir().visit_item_likes_in_module`.
     /// Avoid calling this query directly.
-    query hir_module_items(key: LocalDefId) -> &'tcx hir::ModuleItems {
-        eval_always
+    query hir_module_items(key: LocalDefId) -> rustc_middle::hir::ModuleItems {
+        storage(ArenaCacheSelector<'tcx>)
```

Comment:
What does this line mean? Is changing the cache storage mandatory here, or is it an optimization?

Reply:
If we don't change the cache storage, the query system will clone the return value for each invocation.